numhask/src/NumHask/Algebra/Abstract/Group.hs

```diff
@@ -82,24 +81,24 @@ instance (Associative a, Unital a, Invertible a) => Group a
 --
 -- > a `times` absorb = absorb
 class Magma a =>
-    Absorbing a where
-    absorb :: a
+  Absorbing a where
+  absorb :: a
 
 -- | An Idempotent Magma
 --
 -- > a magma a = a
 class Magma a =>
-      Idempotent a
+  Idempotent a
 
 -- | see http://chris-taylor.github.io/blog/2013/02/25/xor-trick/
 groupSwap :: (Group a) => (a, a) -> (a, a)
 groupSwap (a, b) =
-      let a'  = a `magma` b
-          b'  = a `magma` inv b
-          a'' = inv b' `magma` a'
-      in  (a'', b')
+  let
+    a' = a `magma` b
+    b' = a `magma` inv b
+    a'' = inv b' `magma` a'
+  in (a'', b')
```

Comment:
Btw I don't think this is correct. It needs stronger guarantees than just group, i think.

Also, it seems very randomly placed. 

I know it wasn't introduced here

Reply:
@LeanderK.  Happy to remove - it's just there because I think it's cool.  But what stronger guarantees are needed?  One of the benefits of the numhask style is to find stuff like this that is highly polymorphic, and let ghc get aggressive on it.